src/vivarium_experiment_uk_econ/components/uk_econ.py

```diff
@@ -235,23 +236,13 @@ def setup(self, builder: Builder):
         self.tax_amount = builder.value.register_value_producer('tax_amount', source=self.get_tax_amount)
         self.net_income = builder.value.register_value_producer('net_income', source=self.get_net_income)
 
-        # TODO: load data on income
-        # df = pd.read_excel(fname, # how did I do this previously?
-        log_mean, log_sd = 5, 1
-        def before_tax_income_by_percentile_and_time(x, y):
-            return scipy.stats.lognorm(log_mean, scale=log_sd).ppf(x)
-        self.before_tax_income_by_percentile_and_time = before_tax_income_by_percentile_and_time
-
-        tax_rate = [[0,0],[10,3],[80,13],[85,70],[100,70]]
-        # setup spline for tax rate
-        import scipy.interpolate
-        tax_rate_data = np.array(tax_rate)
-        x = tax_rate_data[:,0]/100
-        y = tax_rate_data[:,1]/100
-        self.tax_rate_func = scipy.interpolate.interp1d(x, y)
-        def after_tax_income_by_percentile_and_time(x, y):
-            return scipy.stats.lognorm(log_mean, scale=log_sd).ppf(x) * (1 - self.tax_rate_func(x))
-        self.after_tax_income_by_percentile_and_time = after_tax_income_by_percentile_and_time
+        # load data on income, make interpolaters
+        self.income_func = {}
+        for when in ['before_tax', 'after_tax']:
+            fname = LOCAL_DATA_DIR.joinpath('income_data.xlsx')
```

Comment:
I strongly recommend making functions in your `data` module like `get_income`.  Keeping data manipulation out of the component makes much more robust and modular code.